---
title: "Single Cell RNA-Seq Analysis"
authors: "Erik Bot, Colombo Lara"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(dplyr)
library(Seurat)
library(patchwork)
library(viridis)
```

# LOAD THE DATA 

Here we load the dataset on Substantia Nigra - Human. 

```{r}
sn.data <- load('./SRA850958_SRS4386112.sparse.RData')

rownames(sm) <- sapply(strsplit(rownames(sm),"_"), `[`, 1)

sn <- CreateSeuratObject(counts = sm, project = "Subnigra", min.cells = 3, min.features = 200)

sn
```


```{r}
head(colnames(sn))
```

# QUALITY CONTROL 

```{r}
# Mithocondrial DNA genes 
grep("^MT-",rownames(sn),value = TRUE)
```

```{r}
# Ribosomal Protein genes
grep("^RP[LS]",rownames(sn),value = TRUE)
```

```{r}
# percentage of mithocondrial DNA and ribosomal protein genes

sn[["percent.mt"]] <- PercentageFeatureSet(sn, pattern = "^MT-")

sn[["percent.rbp"]] <- PercentageFeatureSet(sn, pattern = "^RP[LS]")

```

```{r}
head(sn@meta.data, 5)
```

```{r}
VlnPlot(sn, features = c("nFeature_RNA", "nCount_RNA", "percent.mt","percent.rbp"), ncol = 4)
```

```{r}
VlnPlot(sn, features = c("nFeature_RNA", "nCount_RNA", "percent.mt","percent.rbp"), ncol = 4, pt.size=0, cols = '#e69138')
```

```{r}
VlnPlot(sn, features = "nFeature_RNA", ncol = 1, pt.size=0, y.max = 6000)
```


```{r}
VlnPlot(sn, features = c("percent.mt","percent.rbp"), ncol = 2, pt.size=0, y.max = 20)
```


```{r}
plot1 <- FeatureScatter(sn, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(sn, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")

plot1 + plot2
```

```{r}
plot3 <- FeatureScatter(sn, feature1 = "nCount_RNA", feature2 = "percent.rbp")
plot3
```
Here, we filter the dataset according to the thresholds identified by the previous plots:

```{r}
sn <- subset(sn, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & percent.mt < 5)

sn
```

# DATA NORMALIZATION

```{r}
sn <- NormalizeData(sn, normalization.method = "LogNormalize", scale.factor = 10000)
```

```{r}
sn@assays
```

```{r}
sn@assays$RNA
```

# CELL CYCLE 

```{r}
cc.genes.updated.2019
```

```{r}
sn <- CellCycleScoring(sn, s.features = cc.genes.updated.2019$s.genes, g2m.features = cc.genes.updated.2019$g2m.genes, set.ident = TRUE)

sn[[]]
```

# DIMENSIONALITY REDUCTION 

```{r}
sn <- FindVariableFeatures(sn, selection.method = "vst", nfeatures = 2000)

top10 <- head(VariableFeatures(sn), 10)

plot1 <- VariableFeaturePlot(sn)
plot2 <- LabelPoints(plot = plot1, points = top10, repel = TRUE)
plot2
```

```{r}
all.genes <- rownames(sn)
sn <- ScaleData(sn, features = all.genes)
```

```{r}
sn <- RunPCA(sn, features = VariableFeatures(object = sn))
```
```{r}
print(sn[["pca"]], dims = 1:5, nfeatures = 5)
```

```{r}
VizDimLoadings(sn, dims = 1:3, reduction = "pca", col = '#e69138', nfeatures = 10, ncol = 3)
VizDimLoadings(sn, dims = 4:5, reduction = "pca", col = '#e69138', nfeatures = 10, ncol = 3)
VizDimLoadings(sn, dims = 5, reduction = "pca", col = '#e69138', ncol = 2, nfeatures = 10)
#VizDimLoadings(sn, dims = 4, reduction = "pca", col = '#e69138')
#VizDimLoadings(sn, dims = 5, reduction = "pca", col = '#e69138')
```

```{r}
# checking if they group according to the cell cycle phase
DimPlot(sn, reduction = "pca")
```
## CHOOSING THE BEST DIMENSIONS 

```{r}
# Elbow plot 
ElbowPlot(sn, ndims=20)
```

```{r}
# keeping the PCs that expliain 75 % of variance
pc.touse <- (sn$pca@stdev)^2
pc.touse <- pc.touse/sum(pc.touse)
pc.touse <- cumsum(pc.touse)[1:50]
pc.touse <- min(which(pc.touse>=0.75))
pc.touse
```
### 10 PCs

We test how the analysis proceeds using 10 PCs, according to the elbow plot

```{r}
sn10 <- FindNeighbors(sn, dims = 1:10)

sn10 <- FindClusters(sn10, resolution = 0.5)
```

```{r}
head(Idents(sn10), 5)

table(Idents(sn10))
```
```{r}
head(sn10[[]],5)
```

```{r}
DimPlot(sn, reduction = "pca")
```

```{r}
sn10 <- RunTSNE(sn10, dims=1:10)
DimPlot(sn10, reduction = "tsne", cols = turbo(length(unique(sn10$seurat_clusters))))
```

```{r}
sn10 <- RunUMAP(sn10, dims = 1:10)

DimPlot(sn10, reduction = "umap", cols = turbo(length(unique(sn10$seurat_clusters))))
```

```{r}
VlnPlot(sn10,features="nCount_RNA")
VlnPlot(sn10,features="nFeature_RNA")
VlnPlot(sn10,features="percent.mt")
VlnPlot(sn10,features="percent.rbp")
```

```{r}
library(ggplot2)

sn10@meta.data %>%
  group_by(seurat_clusters,Phase) %>%
  count() %>%
  group_by(seurat_clusters) %>%
  mutate(percent=100*n/sum(n)) %>%
  ungroup() %>%
  ggplot(aes(x=seurat_clusters,y=percent, fill=Phase)) +
  geom_col() +
  ggtitle("Percentage of cell cycle phases per cluster")
```
### 13 PCs

```{r}
sn13 <- FindNeighbors(sn, dims = 1:13)
```

```{r}
sn13 <- FindClusters(sn13, resolution = 0.5)
```

```{r}
head(Idents(sn13), 5)

table(Idents(sn10.2))
```

```{r}
head(sn13[[]],5)
```

```{r}
sn13 <- RunTSNE(sn13, dims=1:13)
DimPlot(sn13, reduction = "tsne", cols = turbo(length(unique(sn13$seurat_clusters))))
```

```{r}
sn13 <- RunUMAP(sn13, dims = 1:13)

DimPlot(sn13, reduction = "umap", cols = turbo(length(unique(sn13$seurat_clusters))))
```

```{r}
VlnPlot(sn13,features="nCount_RNA")
VlnPlot(sn13,features="nFeature_RNA")
VlnPlot(sn13,features="percent.mt")
VlnPlot(sn13,features="percent.rbp")
```

```{r}
library(ggplot2)

sn13@meta.data %>%
  group_by(seurat_clusters,Phase) %>%
  count() %>%
  group_by(seurat_clusters) %>%
  mutate(percent=100*n/sum(n)) %>%
  ungroup() %>%
  ggplot(aes(x=seurat_clusters,y=percent, fill=Phase)) +
  geom_col() +
  ggtitle("Percentage of cell cycle phases per cluster")
```
# CLUSTERS IDENTIFICATION 

We decided to keep 10 PCs, therefore we proceed to identify clusters in this new dimensional space

```{r}
sn10.markers <- FindAllMarkers(sn10, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
```


```{r}
sn10.markers %>%
    group_by(cluster) %>%
    slice_max(n = 5, order_by = avg_log2FC)
```

```{r}
DimPlot(sn10, reduction = "umap", cols = turbo(length(unique(sn10$seurat_clusters))))
```

```{r}
FeaturePlot(sn10, features = c("SLC14A1", "STMN4", "KLK6", "OPALIN", "PTPRZ1"))

FeaturePlot(sn10, features = c("CD74", "MIR219A2", "RGS5", "SYT1", "ITGAX"))
```
MARKER GENES ANALYSIS

```{r}
sn10.markers %>%
    group_by(cluster) %>%
    top_n(n = 10, wt = avg_log2FC) -> top10
DoHeatmap(sn10, features = top10$gene) + NoLegend()
```
We can see similar behaviors in clusters 2 and 3.

```{r}
cluster23.markers <- FindMarkers(sn10, ident.1 = 2, ident.2 = 3, min.pct = 0.25, test.use = "wilcox")
cluster23.markers <- cluster23.markers[order(-cluster23.markers$avg_log2FC),]
head(cluster23.markers, n = 10)
```

Clusters 2 and 3 seem to be different enough.

```{r}
cluster623.markers <- FindMarkers(sn10, ident.1 = 6, ident.2 = c(2,3), min.pct = 0.25, test.use = "wilcox")
cluster623.markers <- cluster623.markers[order(-cluster623.markers$avg_log2FC),]
head(cluster623.markers, n = 10)
```
Clusters 6 against 2 and 3 seem to have the same percentages.


## DECREASE RESOLUTION - 9 CLUSTERS


```{r}
sn10.2 <- FindNeighbors(sn, dims = 1:10)
sn10.2 <- FindClusters(sn10.2, resolution = 0.175)

sn10.2 <- RunUMAP(sn10.2, dims = 1:10)

DimPlot(sn10.2, reduction = "umap", cols = turbo(length(unique(sn10.2$seurat_clusters))))
```
### MARKER GENES ANALYSIS AND CLUSTER IDENTIFICATION 

```{r}
sn10.2.markers <- FindAllMarkers(sn10.2, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
```

```{r}
sn10.2.markers %>%
    group_by(cluster) %>%
    slice_max(n = 5, order_by = avg_log2FC)
```

```{r}
FeaturePlot(sn10.2, features = c("CLDN11", "SLC14A1", "STMN4", "PTPRZ1", "CD74"))

FeaturePlot(sn10.2, features = c("CERCAM", "RGS5", "SYT1", "ITGAX"))
```

We can see that cluster 5 has genes that are not so specific for its cluster and overlap also cluster 0 and 2. We can also see that cluster 4 and 8 have overlaps.

```{r}
sn10.2.markers %>%
    group_by(cluster) %>%
    top_n(n = 5, wt = avg_log2FC) -> top5
DoHeatmap(sn10.2, features = top5$gene) + NoLegend()
```

We can see that 0 has similar patterns to clusters 2 and 5, which is expected.


```{r}
cluster502.markers <- FindMarkers(sn10.2, ident.1 = 5, ident.2 = c(0,2), min.pct = 0.25, test.use = "wilcox")
cluster502.markers <- cluster502.markers[order(-cluster502.markers$avg_log2FC),]
cluster502.markers <- cluster502.markers[which(cluster502.markers$avg_log2FC > 0),]
head(cluster502.markers, n = 20)
length(cluster502.markers$avg_log2FC)
```

No particularly significant genes.

```{r}
cluster52.markers <- FindMarkers(sn10.2, ident.1 = 5, ident.2 = 2, min.pct = 0.25, test.use = "wilcox")
cluster52.markers <- cluster52.markers[order(-cluster52.markers$avg_log2FC),]
length(cluster52.markers$avg_log2FC)

cluster52.markers <- cluster52.markers[which(cluster52.markers$pct.1 > cluster52.markers$pct.2),]
cluster52.markers <- cluster52.markers[which(cluster52.markers$avg_log2FC > 0),]
length(cluster52.markers$avg_log2FC)

head(cluster52.markers, 20)
```

Some genes of difference between cluster 0 and 5, not such a relevant functional implication.

```{r}
cluster50.markers <- FindMarkers(sn10.2, ident.1 = 5, ident.2 = 0, min.pct = 0.25, test.use = "wilcox")
cluster50.markers <- cluster50.markers[order(-cluster50.markers$avg_log2FC),]
head(cluster50.markers, n = 30)
length(cluster50.markers$avg_log2FC)
```

No differences.

```{r}
cluster05.markers <- FindMarkers(sn10.2, ident.1 = 0, ident.2 = 5, min.pct = 0.25, test.use = "wilcox")
cluster05.markers <- cluster05.markers[order(-cluster05.markers$avg_log2FC),]
head(cluster05.markers, n = 30)
length(cluster05.markers$avg_log2FC)
```

Relevant gene differences: MAG, CRYAB, CNP

```{r}
cluster25.markers <- FindMarkers(sn10.2, ident.1 = 2, ident.2 = 5, min.pct = 0.25, test.use = "wilcox")
cluster25.markers <- cluster25.markers[order(-cluster25.markers$avg_log2FC),]
head(cluster25.markers, n = 30)
length(cluster25.markers$avg_log2FC)
```

Relevant gene differences: CRYAB, MAG, TPPP, S100B

Hypothesis is that cluster 5 is composed of oligodendrocytes that still have to differentiate into satellite and parafascicular. Pre-myelinating.

```{r}
cluster025.markers <- FindMarkers(sn10.2, ident.1 = c(0,2), ident.2 = 5, min.pct = 0.25, test.use = "wilcox")
cluster025.markers <- cluster025.markers[order(-cluster025.markers$avg_log2FC),]
head(cluster025.markers, n = 30)
length(cluster025.markers$avg_log2FC)
```

Genes: S100B, CRYAB, MAG, ERMN, CNP, TPPP3, PLP, PLPP2 --> myelinization factors.

```{r}
FeaturePlot(sn10.2, features = c("MAG"))
```
Differences between 0 and 2

```{r}
cluster02.markers <- FindMarkers(sn10.2, ident.1 = 0, ident.2 = 2, min.pct = 0.25, test.use = "wilcox")
cluster02.markers <- cluster02.markers[order(-cluster02.markers$avg_log2FC),]
head(cluster02.markers, n = 30)
length(cluster02.markers$avg_log2FC)
```

Cluster 0 has PTGDS, GAPDH, CLDN11, QDPR, OPALIN overexpressed --> methabolic --> satellite

```{r}
cluster20.markers <- FindMarkers(sn10.2, ident.1 = 2, ident.2 = 0, min.pct = 0.25, test.use = "wilcox")
cluster20.markers <- cluster20.markers[order(-cluster20.markers$avg_log2FC),]
head(cluster20.markers, n = 30)
length(cluster20.markers$avg_log2FC)
```

Cluster 2 has no particular overexpression.

They are all oligo:

```{r}
cluster0AND2AND5.markers <- FindMarkers(sn10.2, ident.1 = c(0,2,5), min.pct = 0.25, test.use = "wilcox")
cluster0AND2AND5.markers <- cluster0AND2AND5.markers[order(-cluster0AND2AND5.markers$avg_log2FC),]
head(cluster0AND2AND5.markers, n = 10)
```

Check 4 against 8 and 1 against 8

```{r}
cluster48.markers <- FindMarkers(sn10.2, ident.1 = 4, ident.2 = 8, min.pct = 0.25, test.use = "wilcox")
cluster48.markers <- cluster48.markers[order(-cluster48.markers$avg_log2FC),]
head(cluster48.markers, n = 30)
length(cluster48.markers$avg_log2FC)
```

```{r}
cluster84.markers <- FindMarkers(sn10.2, ident.1 = 8, ident.2 = 4, min.pct = 0.25, test.use = "wilcox")
cluster84.markers <- cluster84.markers[order(-cluster84.markers$avg_log2FC),]
head(cluster84.markers, n = 30)
length(cluster84.markers$avg_log2FC)
```


```{r}
cluster8.markers <- FindMarkers(sn10.2, ident.1 = 8, min.pct = 0.25, test.use = "wilcox")
cluster8.markers <- cluster8.markers[order(-cluster8.markers$avg_log2FC),]
head(cluster8.markers, n = 30)
length(cluster8.markers$avg_log2FC)
```

Cluster 8 markers are ITGAX, LPAR6, ZFP36L1, LRRK1, CSF1R --> microglial cells


```{r}
cluster4.markers <- FindMarkers(sn10.2, ident.1 = 4,  min.pct = 0.25, test.use = "wilcox")
cluster4.markers <- cluster4.markers[order(-cluster4.markers$avg_log2FC),]
head(cluster4.markers, n = 30)
length(cluster4.markers$avg_log2FC)
```
Again, microglial cells.

```{r}
cluster4AND8.markers <- FindMarkers(sn10.2, ident.1 = c(4,8), min.pct = 0.25, test.use = "wilcox")
cluster4AND8.markers <- cluster4AND8.markers[order(-cluster4AND8.markers$avg_log2FC),]
head(cluster4AND8.markers, n = 10)
```



```{r}
cluster18.markers <- FindMarkers(sn10.2, ident.1 = 1, ident.2 = 8, min.pct = 0.25, test.use = "wilcox")
cluster18.markers <- cluster18.markers[order(-cluster18.markers$avg_log2FC),]
head(cluster18.markers, n = 10)
length(cluster18.markers$avg_log2FC)
```

```{r}
cluster81.markers <- FindMarkers(sn10.2, ident.1 = 8, ident.2 = 1, min.pct = 0.25, test.use = "wilcox")
cluster81.markers <- cluster81.markers[order(-cluster81.markers$avg_log2FC),]
head(cluster81.markers, n = 10)
length(cluster81.markers$avg_log2FC)
```


We can see that clusters 4 and 1 seem to have no differentially expressed genes compared to 8, while 8 has some specific genes that differ from both cluster 4 and cluster 1.

Check cluster 8 vs 1 and 4

```{r}
cluster814.markers <- FindMarkers(sn10.2, ident.1 = 8, ident.2 = c(1,4), min.pct = 0.25, test.use = "wilcox")
cluster814.markers <- cluster814.markers[order(-cluster814.markers$avg_log2FC),]
head(cluster814.markers, n = 30)
length(cluster814.markers$avg_log2FC)
```

The specific genes found seem not to be that specific. It seems that this cluster is composed of microglia cells, but with some more specific genes.

```{r}
DimPlot(sn10.2, reduction = "umap", cols = turbo(length(unique(sn10.2$seurat_clusters))))
```

```{r}
sn10.2.markers %>%
    group_by(cluster) %>%
    slice_max(n = 5, order_by = avg_log2FC)
```
Cluster 0: CLDN11 -> satellite oligodendrocytes
Cluster 1: SLC14A1 -> astrocytes
Cluster 2: S100B -> interfascicular oligodendrocytes
Cluster 3: PTPRZ1 -> oligodendrocytes precursor cells
Cluster 4: LPAR6 -> microglia cells
Cluster 5: CERCAM -> pre-myelinating oligodendrocytes
Cluster 6: RGS5 -> pancreatic stellate cells
Cluster 7: SYT1 -> neurons
Cluster 8: ITGAX -> microglia cells

```{r}
VlnPlot(sn10.2, features = c("CLDN11", "SLC14A1")) # clusters 0; cluster 1
VlnPlot(sn10.2, features = c("S100B", "PTPRZ1")) # cluster 2; cluster 3
VlnPlot(sn10.2, features = c("LPAR6", "CERCAM")) # cluster 4; cluster 5
VlnPlot(sn10.2, features = c("RGS5", "SYT1")) # cluster 6; cluster 7
VlnPlot(sn10.2, features = c("ITGAX")) # cluster 8
```

We can see that cluster 8 has many overlapping with the other clusters, so we might want to merge it with cluster 4.

```{r}
FeaturePlot(sn10.2, features = c("CLDN11", "SLC14A1", "S100B", "PTPRZ1", "LPAR6"))

FeaturePlot(sn10.2, features = c("CERCAM", "RGS5", "SYT1", "ITGAX"))
```


```{r}
new.cluster.ids <- c("Satellite Oligodendrocytes", "Astrocytes", "Interfascicular Oligodendrocytes", "Oligodendrocytes progenitor cells", "Active Microglia cells", "Pre-myelinating Oligodendrocytes", "Pancreatic stellar cells", "Neurons", "Ramified Microglia cells")
names(new.cluster.ids) <- levels(sn10.2)
sn10.2 <- RenameIdents(sn10.2, new.cluster.ids)
DimPlot(sn10.2, reduction = "umap", label = TRUE, pt.size = 0.5, label.size = 6) + NoLegend()
```

## DECREASING RESOLUTION - 7 clusters

```{r}
sn10.3 <- FindNeighbors(sn, dims = 1:10)
sn10.3 <- FindClusters(sn10.3, resolution = 0.05)

sn10.3 <- RunUMAP(sn10.3, dims = 1:10)

DimPlot(sn10.3, reduction = "umap", cols = turbo(length(unique(sn10.3$seurat_clusters))))
```
### MARKER GENES ANALYSIS AND CLUSTER IDENTIFICATION 

```{r}
sn10.3.markers <- FindAllMarkers(sn10.3, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
```

```{r}
sn10.3.markers %>%
    group_by(cluster) %>%
    slice_max(n = 5, order_by = avg_log2FC)
```

```{r}
FeaturePlot(sn10.2, features = c("MBP", "CLDN11", "SLC14A1", "LPAR6", "PTPRZ1", "RGS5", "SYT1"))
```

```{r}
sn10.3.markers %>%
    group_by(cluster) %>%
    top_n(n = 10, wt = avg_log2FC) -> top10
DoHeatmap(sn10.3, features = top10$gene) + NoLegend()
```

We can see that now all clusters are different from each other, a part from cluster 0 and 1, that share some similarity. We can check them.

```{r}
cluster01.markers <- FindMarkers(sn10.3, ident.1 = 0, ident.2 = 1, min.pct = 0.25, test.use = "wilcox")
cluster01.markers <- cluster01.markers[order(-cluster01.markers$avg_log2FC ),]
head(cluster01.markers, n = 10)

#length(cluster01.markers$avg_log2FC) # 562 differentially expressed genes

#sum(startsWith(rownames(cluster01.markers), 'MT-')) # 11 mitochondrial genes (2%)
```

```{r}
cluster01.markers.p <- cluster01.markers[which(cluster01.markers$avg_log2FC > 0),]
cluster01.markers.p <- cluster01.markers.p[which(cluster01.markers.p$pct.1 > 0.25),]
cluster01.markers.p <- cluster01.markers.p[which(!startsWith(rownames(cluster01.markers.p), 'RP')),]
cluster01.markers.p <- cluster01.markers.p[order(-cluster01.markers.p$avg_log2FC),]

head(cluster01.markers.p, 20)

cluster01.markers.p[which(startsWith(rownames(cluster01.markers.p), 'CRYAB')),]

```


```{r}
cluster10.markers <- FindMarkers(sn10.3, ident.1 = 1, ident.2 = 0, min.pct = 0.25, test.use = "wilcox")
cluster10.markers <- cluster10.markers[order(-cluster10.markers$avg_log2FC),]
head(cluster10.markers, n = 10)
length(cluster10.markers$avg_log2FC)

cluster10.markers[which(startsWith(rownames(cluster10.markers), 'TPPP')),]

cluster10.markers[which(startsWith(rownames(cluster10.markers), 'PLP')),]
```

```{r}
cluster10.markers.p <- cluster10.markers[which(cluster10.markers$avg_log2FC > 0),]
cluster10.markers.p <- cluster10.markers.p[which(cluster10.markers.p$pct.1 > 0.25),]
cluster10.markers.p <- cluster10.markers.p[which(!startsWith(rownames(cluster10.markers.p), 'RP')),]
cluster10.markers.p <- cluster10.markers.p[order(-cluster10.markers.p$avg_log2FC),]

head(cluster10.markers.p, 20)
```


We can see that cluster 1 has a lot of differentially overexpressed genes compared to 0, but 0 has no particular differentially overexpressed gene from 1. They might be of the same cell type, but 1 a more speicifc subtype.

```{r}
cluster1AND0.markers <- FindMarkers(sn10.3, ident.1 = c(1,0), min.pct = 0.25, test.use = "wilcox")
cluster1AND0.markers <- cluster1AND0.markers[order(-cluster1AND0.markers$avg_log2FC),]
head(cluster1AND0.markers, n = 10)
```


```{r}
DimPlot(sn10.3, reduction = "umap", cols = turbo(length(unique(sn10.3$seurat_clusters))))
```

```{r}
VlnPlot(sn10.3, features = c("MBP", 'PLP1', "GAPDH")) # clusters 0 and 1; cluster 1
VlnPlot(sn10.3, features = c("SLC14A1", "LPAR6")) # cluster 2; cluster 3
VlnPlot(sn10.3, features = c("PTPRZ1", "RGS5", "SYT1")) # cluster 4; cluster 5; cluster 6
```

```{r}
DotPlot(sn10.3, features = c("MBP", "PLP1", "GAPDH", "SLC14A1", "LPAR6", "PTPRZ1", "RGS5", "SYT1")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) 
```
We can see that all the genes highlight different clusters, while MBP is specific for cluster 0 and 1, but has an higher expression in cluster 0.

```{r}
sn10.3.cc <- CellCycleScoring(sn10.3, s.features = cc.genes.updated.2019$s.genes, g2m.features = cc.genes.updated.2019$g2m.genes, set.ident = TRUE)

DimPlot(sn10.3.cc, reduction = "umap", cols = viridis(3))
```
There is no apparent impact of the cell cycle.

```{r}
sn10.3.markers %>%
    group_by(cluster) %>%
    slice_max(n = 5, order_by = avg_log2FC)
```

```{r}
print(sn[["pca"]], dims = 1:5, nfeatures = 5)
```

0-MBP: Interfascicular oligodendrocytes

1-GAPDH: Oligodendrocytes
1-CLDN11: Oligodendrocytes
1- TPPP: Myelinating Oligodendrocytes

2-SLC14A1: Astrocytes

3-LPAR6: Microglia cells

4-PTPRZ1: Oligodendrocytes progenitor cells

5-RGS5: Pancreatic stellate cells

6-SYT1: Neurons

```{r}
#new.cluster.ids <- c("Interfascicular Oligodendrocytes", "Satellite Oligodendrocytes", "Astrocytes", "Microglia cells", "Oligodendrocytes progenitor cells", "Pancreatic stellate cells", "Neurons")
#names(new.cluster.ids) <- levels(sn10.3)
#sn10.3 <- RenameIdents(sn10.3, new.cluster.ids)
DimPlot(sn10.3, reduction = "umap", cols = turbo(length(unique(sn10.3$seurat_clusters))), label = TRUE, pt.size = 0.5) 
```
```{r}
freq.matrix <- matrix(0, nrow = max(as.integer(sn10.df$clusters)) + 1, ncol = max(as.integer(sn13.df$clusters) +1)) 

freq <- as.data.frame(freq.matrix) 

rownames(freq) <- as.character(seq(0, max(as.integer(sn10.df$clusters)))) 

colnames(freq) <- as.character(seq(0, max(as.integer(sn13.df$clusters)))) 

for(n in seq(0, max(as.integer(sn13.df$clusters)))) { 
for(i in seq(0, max(as.integer(sn10.df$clusters)))) { 
for(el in freq.df[,1]) { 
if(startsWith(el, as.character(i)) && endsWith(el, as.character(n))) { 
freq[as.character(i),as.character(n)] <- as.integer(freq.df[el, 2]) 
} 
} 
} 
} 

freq 

heatmap(log10(as.matrix(freq) + 1), Colv = NA, Rowv = NA, main = 'Log10 number of shared cells', xlab = 'SN10 clusters', ylab = 'SN13 clusters') 
```

```{r}
DotPlot(sn10.2, features = c("CLDN11", "TUBB2B", "CERCAM", "MBP", "MAG", "PTPRZ1", "RGS5", "SYT1", "ITGAX", "SLC14A1")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))

DotPlot(sn10.2, features = c("PTPRZ1", "RGS5", "SYT1")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) 

DotPlot(sn10.2, features = c("ITGAX", "SLC14A1", "LPAR6", "DST")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
```

# PCs INTERSECTION

```{r}
sn10.2 <- FindNeighbors(sn, dims = 1:10)
sn10.2 <- FindClusters(sn10.2, resolution = 0.175)

sn10.2 <- RunUMAP(sn10.2, dims = 1:10)
sn13.df <- as.data.frame(Idents(sn13))

colnames(sn13.df) <- 'clusters'

sn13.df$clusters <- as.character(sn13.df$clusters)

sn10.2.df <- as.data.frame(Idents(sn10.2))

colnames(sn10.2.df) <- 'clusters'

sn10.2.df$clusters <- as.character(sn10.2.df$clusters)
```

```{r}
sum(rownames(sn10.2.df)!=rownames(sn13.df))
```

```{r}
df.inter <- sn10.2.df

colnames(df.inter) <- 'cluster 10'

df.inter['cluster 13'] <- sn13.df$clusters

df.inter

df.inter$'couple' <- 0

for (n in seq(1, nrow(df.inter))){
  df.inter[n, 3] <- paste(df.inter[n, 1], (df.inter[n,2]), sep = "-")
}

freq.df <- as.data.frame(table(df.inter[3]))

rownames(freq.df) <- freq.df[,1]

colnames(freq.df) <- c('couple', 'freq')

barplot(freq.df[,2], names.arg = freq.df[,1], las = 2 )

```

```{r}
freq.matrix <- matrix(0, nrow = max(as.integer(sn10.2.df$clusters)) + 1, ncol = max(as.integer(sn13.df$clusters) +1))

freq <- as.data.frame(freq.matrix)

rownames(freq) <- as.character(seq(0, max(as.integer(sn10.2.df$clusters))))

colnames(freq) <- as.character(seq(0, max(as.integer(sn13.df$clusters))))

for(n in seq(0, max(as.integer(sn13.df$clusters)))) {
  for(i in seq(0, max(as.integer(sn10.2.df$clusters)))) {
    for(el in freq.df[,1]) {
      if(startsWith(el, as.character(i)) && endsWith(el, as.character(n))) {
        freq[as.character(i),as.character(n)] <- as.integer(freq.df[el, 2])
      }
    }
  }
}

freq
```

```{r}
heatmap(log10(as.matrix(freq) + 1), Colv = NA, Rowv = NA, main = 'Log10 number of shared cells', xlab = 'SN13 clusters', ylab = 'SN10 clusters')
```
